gtsfm/loader/one_d_sfm_loader.py

```diff
@@ -0,0 +1,162 @@
+
+Authors: Yanwei Du
+
+"""
+
+import glob
+import numpy as np
+import os
+from pathlib import Path
+from typing import List, Optional
+
+import scipy.io
+from gtsam import Cal3Bundler, Pose3
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.verification as verification_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class OneDSFMLoader(LoaderBase):
+    def __init__(
+        self,
+        folder: str,
+        image_extension: str = "jpg",
+        max_resolution: int = 640,
+        max_num_imgs: int = 0
+    ) -> None:
+        """Initializes to load from a specified folder on disk.
+
+        Args:
+            folder: the base folder which contains image sequence
+            image_extension: file extension for the image files. Defaults to 'jpg'.
+            max_resolution: integer representing maximum length of image's short side, i.e.
+                the smaller of the height/width of the image. e.g. for 1080p (1920 x 1080),
+                max_resolution would be 1080. If the image resolution max(height, width) is
+                greater than the max_resolution, it will be downsampled to match the max_resolution.
+            max_num_imgs: integer representing max number of images to process in the sequence,
+                0 or negative values means to process all the images
+        """
+        super().__init__(max_resolution=max_resolution)
+
+        # fetch all the file names in /images folder
+        search_path = os.path.join(folder, 'images', f"*.{image_extension}")
+
+        # @NOTE for now we only process images with valid EXIF data
+        self._image_paths, self._num_exif_imgs, self._num_all_imgs = \
+            self.get_images_with_valid_exif(search_path, max_num_imgs)
+        self._num_imgs = len(self._image_paths)
+
+        # self._image_paths.sort()
+        print(f'selected {self._num_imgs} out of {self._num_exif_imgs} images with valid exif')
+
+        if self._num_imgs == 0:
+            raise RuntimeError(
+                f"Loader could not find any images with the specified file extension in {search_path}")
+
+    def get_images_with_valid_exif(self, search_path: str, max_num_imgs: int) -> List[str]:
+        """Return a subset of images with valid exif.
+
+        Args:
+            search_path: image sequence search path
+            max_num_imgs: the maximum number of images to process
+
+        Returns:
+            list of selected image path.
+            the number of images with valid exif data.
+            the number of all the images.
+        """
+
+        all_image_paths = glob.glob(search_path)
+        num_all_imgs = len(all_image_paths)
+        exif_image_paths = []
+        for p in all_image_paths:
+            # drop images without valid exif data
+            if io_utils.load_image(p).get_intrinsics_from_exif():
+                exif_image_paths.append(p)
+
+        # randomly select a subset to process
```

Comment:
nit capitalization and punctuation: `Randomly select a subset to process.`

Reply:
Done.